components/support/rs_client/src/rs_client.udl

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace rs_client {};
+
+[Custom]
+typedef string RsJsonObject;
+
+dictionary RemoteSettingsConfig {
+    string? server_url;
+    string? bucket_name;
+    string collection_name;
+};
+
+dictionary RemoteSettingsResponse {
+    sequence<Record> records;
+    u64 last_modified;
+};
+
+dictionary Record {
+    string id;
+    u64 last_modified;
+    Attachment? attachment;
+    RsJsonObject fields;
+};
+
+dictionary Attachment {
+    string filename;
+    string mimetype;
+    string location;
+    string hash;
+    u64 size;
+};
```

Comment:
Should I write comments here for these types? Will they propagate through the `extern` references in the top level components?